src/Dapr.Client/ServiceInvocationRequest.cs

```diff
@@ -0,0 +1,69 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Client
+{
+    using Dapr.Client;
+    using Dapr.Client.Http;
+
+    /// <summary>
+    /// Represents an Invoke Request used for service invocation
+    /// </summary>
+    /// <typeparam name="TRequest">Data type of the request.</typeparam>
+    public sealed class ServiceInvocationRequest<TRequest>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServiceInvocationRequest{TRequest}"/> class.
+        /// </summary>
+        public ServiceInvocationRequest()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServiceInvocationRequest{TRequest}"/> class.
+        /// </summary>
+        /// <param name="appId">The app identifier.</param>
+        /// <param name="methodName">The name of the method to invoke.</param>
+        /// <param name="body">The request body.</param>
+        /// <param name="contentType">The content type.</param>
+        /// <param name="httpExtension">HTTP extension info (optional).</param>
+        public ServiceInvocationRequest(string appId, string methodName, TRequest body, string contentType, HTTPExtension httpExtension)
+        {
+            ArgumentVerifier.ThrowIfNull(appId, nameof(appId));
+            ArgumentVerifier.ThrowIfNull(methodName, nameof(methodName));
+
+            this.AppId = appId;
+            this.MethodName = methodName;
+            this.Body = body;
+            this.ContentType = contentType;
+            this.HttpExtension = httpExtension;
+        }
```

Comment:
As general design feedback, it's better to stick to one design pattern. If the class accepts a bunch of constructor arguments, I usually expect it to be immutable, or to have those fields immutable.

It's a little off to have a no-arg constructor and one that validates. Having something show up in the constructor (vs a property) doesn't really add convenience - both are easy to use.